### PR TITLE
Template misc

### PR DESCRIPTION
**Pull request by b0r6**
* Added template that's set to 'openvdc'
* The local Image test checks which template was used for creating the container and fails if it's not 'lxc-openvdc'

### CODE REVIEW

File: ci/citest/acceptance-test/tests/local_image_test.go [hunk 1]
```diff
@@ -3,20 +3,46 @@
...
+
+        if err != nil {
+                t.Error(err)
+        }
+
+        t.Log(stdout.String())
+
 	// Use custom lxc-template.
-	stdout, _ := RunCmdAndReportFail(t, "openvdc", "run", "centos/7/lxc", `{"lxc_template":{"template":"openvdc"}, "node_groups":["linuxbr"]}`)
+	stdout, _ = RunCmdAndReportFail(t, "openvdc", "run", "https://raw.githubusercontent.com/axsh/openvdc/template-misc/templates/centos/7/lxc2.json", `{"node_groups":["linuxbr"]}`)
```
Comment (unakatsuo) [inline ci/citest/acceptance-test/tests/local_image_test.go]: when this branch gets removed after merging, the test will fail. lxc2.json should be bundled as test fixtures.

Please refer:
- https://github.com/axsh/openvdc/tree/template-misc/ci/citest/acceptance-test/tests/fixtures
- https://github.com/axsh/openvdc/blob/template-misc/ci/citest/acceptance-test/tests/cmd_template_test.go#L11-L17

File: cmd/lxc-openvdc/main.go [hunk 2]
```diff
@@ -179,7 +179,7 @@ func GetFile(fileName string) error {
 	}
 
 	if res.StatusCode != 200 {
-		return errors.New(fmt.Sprintf("Http status code: %s", res.StatusCode))
+		return errors.New(fmt.Sprintf("Url: %s Http status code: %s", downloadUrl, res.StatusCode))
```
Comment (unakatsuo) [inline cmd/lxc-openvdc/main.go]: errors.New() -> errors.Errorf()

https://github.com/pkg/errors/blob/master/errors.go#L111

File: ci/citest/acceptance-test/tests/local_image_test.go [hunk 3]
```diff
@@ -3,20 +3,46 @@
...
+	}
+	if stdout.Len() == 0 {
+		t.Errorf("Couldn't read %s", configFile)
+	}
+
+	t.Log(stdout.String())
+
+	s := strings.Split(strings.TrimSpace(stdout.String()), "/")
+	templateUsed := s[len(s)-1]
+	if templateUsed == "" || templateUsed != "lxc-openvdc" {
```
Comment (unakatsuo) [inline ci/citest/acceptance-test/tests/local_image_test.go]: ``templateUsed == ""`` is duplicate condition.

File: ci/citest/acceptance-test/tests/local_image_test.go [hunk 4]
```diff
@@ -3,20 +3,46 @@
...
+	stdout, _, err = RunSsh(executor_lxc_ip, fmt.Sprintf("echo | sudo head -n 1 %s", configFile))
+
+	if err != nil {
+		t.Error(err)
+	}
+	if stdout.Len() == 0 {
+		t.Errorf("Couldn't read %s", configFile)
+	}
+
+	t.Log(stdout.String())
```
Comment (unakatsuo) [inline ci/citest/acceptance-test/tests/local_image_test.go]: Better to check if verbose flag is passed from command line when you put non-error messages.

```
if testing.Verbose() {
    t.Log(stdout.String())
}
```

**Review by unakatsuo — CHANGES_REQUESTED** [review]

File: cmd/lxc-openvdc/main.go [hunk 5]
```diff
@@ -179,7 +179,7 @@ func GetFile(fileName string) error {
 	}
 
 	if res.StatusCode != 200 {
-		return errors.New(fmt.Sprintf("Url: %s Http status code: %s", downloadUrl, res.StatusCode))
+		return errors.Errorf(fmt.Sprintf("Url: %s Http status code: %s", downloadUrl, res.StatusCode))
```
Comment (unakatsuo) [inline cmd/lxc-openvdc/main.go]: you can remove ``fmt.Sprintf``

**Review by unakatsuo — CHANGES_REQUESTED** [review]

File: cmd/lxc-openvdc/main.go (hunk 5 above)
Comment (b0r6) [reply cmd/lxc-openvdc/main.go]: Ah, missed that.

**Review by unakatsuo — APPROVED** [review]

**Comment by unakatsuo** [comment]
For the future action,  ``lxc_template`` JSON field should allow the URI field for the base image if ``lxc-openvc`` is set.

```json
 "lxc_template": {
     "openvdc": {
         "image_uri": "https://example.com/centos7.x86_46.lxc.img.tgz",
         "sha256_sum": "abcdefabcedfabcedfabcedf"
      }
}
```

**Pull request merged**